File: python/131-140/wordBreak.py
    def wordBreak(self, s, wordDict):
        if s == None or len(s)==0:
            return true
        size = len(s)
        d = [False for i in xrange(size+1)]
        d[0] = True
        for i in xrange(1,size+1):
            for k in xrange(i):
                if d[k] and s[k:i] in wordDict:
                    d[i] = True
        return d[size]

File: Cpp/130-140/word Break.cpp
class Solution {
public:
  //o(n^2) time   
  // 12ms
  // 4 error   2 write fault object , 1 substr index error , 1 count!=0 forget
    bool wordBreak(string s, unordered_set<string>& wordDict) {
      int size = s.size();
      if(size == 0)
        return true;
      vector<bool> base(size+1,false);  //error 1   vector<int>
      base[0] = true;
      for(int i=0; i<s.size(); i++){ // error 4 wordDict.size()
        for(int j=0; j<i+1; j++)
          if(base[j]&&wordDict.count(s.substr(j,i+1-j))!=0){   //error 2,3  wordDict.count(s.substr(j,i+1));
            base[i+1] = true;
            break;
          }
      }
      return base[size];
    }
};
